src/android/usb_host/usb_device.h

```diff
@@ -0,0 +1,98 @@
+/* License: Apache 2.0. See LICENSE file in root directory. */
+/* Copyright(c) 2019 Intel Corporation. All Rights Reserved. */
+//
+// Created by daniel on 10/16/2018.
+//
+
+#pragma once
+
+#include "usbhost.h"
+#include "usb_configuration.h"
+#include "usbhost.h"
+#include "usb_pipe.h"
+
+#include <string>
+#include <vector>
+#include <regex>
+#include <iostream>
+#include <random>
+#include <memory>
+#include <sstream>
+#include <thread>
+
+namespace librealuvc
+{
+    namespace usb_host
+    {
+        class device {
+        private:
+            static std::string toString(char *cstr);
+            static std::string descriptorTypeToString(uint8_t dt);
+            static std::string classToString(uint8_t cls);
+
+            bool _pull_requests = true;
+            std::unordered_map<int, std::shared_ptr<usb_pipe>> _pipes;
+            std::shared_ptr<std::thread> _pull_thread;
+
+        protected:
+            usbhost_device *_handle;
```

Comment:
Could we make this a shared pointer? It seems a few things hold a pointer to this `usbhost_device`, it's not obvious where it's supposed to get deleted again.

Reply:
Yeah, it'd be nice. It's used in _way_ too many places right now, and it requires a thorough investigation on how and where exactly it is used. I just spent a few minutes looking around, and never seen it being deleted anywhere 😬